pkg/plugins/eventing/sns/sns.go

```diff
@@ -17,18 +17,19 @@ package sns_service
 import (
```

Comment:
I am wondering whether it would be useful to add an addition level of hierarchy here to support different plugin implementations from the same vendor. 

Not sure if perfect example but you get the picture:
```
eventing/aws/sns/sns.go
eventing/aws/sqs/sqs.go
```

Reply:
providers are intentionally left out of the hierarchy, they were originally there but were more pain than it was worth.

With the current hierarchy `eventing/sns/sns.go` and `eventing/sqs/sqs.go` also work without adding the extra step. Theres always the option of including the base provider to create a fully qualified product name i.e. `aws_sns`.